clients/ruby/cac-client/cacclient.rb

```diff
@@ -0,0 +1,91 @@
+require 'ffi'
+require 'os'
+require 'dotenv'
+
+Dotenv.load
+
+module MergeStrategy
+    REPLACE = "REPLACE"
+    MERGE = "MERGE"
+end
+
+module CACLib
+    extend FFI::Library
+
+    extension = "dylib"
+    if OS.windows?
+        extension = "dll"
+    elsif OS.mac?
+        extension = "dylib"
+    else
+        extension = "so"
+    end
+    libpath = File.expand_path(File.join(__dir__, "#{ENV["SUPERPOSITION_LIB_PATH"]}/libcac_client.#{extension}"))
+    ffi_lib libpath
+
+    class ArcClient < FFI::Struct
+        layout :dummy, :int
+    end
+
+    # Attach C functions
+    attach_function :cac_last_error_length, [], :int
+    attach_function :cac_last_error_message, [], :string
+    attach_function :cac_free_string, [:pointer], :void
+    attach_function :cac_new_client, [:string, :ulong, :string], :int
+    attach_function :cac_start_polling_update, [:string], :void
+    attach_function :cac_free_client, [ArcClient.by_ref], :void
+    attach_function :cac_get_client, [:string], ArcClient.by_ref
+    attach_function :cac_get_last_modified, [ArcClient.by_ref], :string
+    attach_function :cac_get_config, [ArcClient.by_ref, :string, :string], :string
+    attach_function :cac_get_resolved_config, [ArcClient.by_ref, :string, :string, :string], :string
+    attach_function :cac_get_default_config, [ArcClient.by_ref, :string], :string
```

Comment:
Can we improve the return type for the functions like cac_get_config, cac_get_resolved_config and others to respective types like map .....
Refer haskell client.